metrics/sinks/stackdriver/stackdriver.go

```diff
@@ -327,13 +327,25 @@ func CreateStackdriverSink(uri *url.URL) (core.DataSink, error) {
 		return nil, err
 	}
 
-	cluster_name := ""
+	clusterName := ""
 	if len(opts["cluster_name"]) >= 1 {
-		cluster_name = opts["cluster_name"][0]
+		clusterName = opts["cluster_name"][0]
 	} else {
 		glog.Warning("Cluster name required but not provided, using empty cluster name.")
 	}
 
+	if clusterName == "" {
+		glog.Info("An empty cluster name has been provided, checking the GCE Metadata Server to try to auto-detect.")
+
+		clusterName_, err := gce.InstanceAttributeValue("cluster-name")
```

Comment:
Shouldn't we perform whole check inside gce.OnGCE block below?

Reply:
Updated